### PR TITLE
Add notes about primary and dependent apps

### LINKED ISSUE #5: Document how to register and use primary and dependent (sub) apps
Coming from [this GitLab issue (private)](https://gitlab.cc-asp.fraunhofer.de/MarketPlace/platform/issue-reporting/-/issues/83).

### PR DESCRIPTION
**Pull request by pablo-de-andres**
Closes #5

### CODE REVIEW

**Comment by pablo-de-andres** [comment]
@flongford want to take a quick look to see if this adds some meaningful information?
Note that this will be revisited once the changes in this [(private) GitLab issue](https://gitlab.cc-asp.fraunhofer.de/MarketPlace/platform/issue-reporting/-/issues/111) have been addressed.

File: docs/source/apps/sub_apps.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Primary and dependent applications
+
+MarketPlace supports the registration of applications contained within another platform, such as JupyterHub-based [AiiDalab](https://aiidalab.readthedocs.io/) and [SimPhony remote](https://simphony-remote.readthedocs.io/).
```
Comment (flongford) [inline docs/source/apps/sub_apps.md]: Nit: minor amendment just for clarity
```suggestion
MarketPlace supports the registration of one or more applications contained within another platform, such as JupyterHub-based [AiiDalab](https://aiidalab.readthedocs.io/) and [SimPhony remote](https://simphony-remote.readthedocs.io/).
```

**Review by flongford — APPROVED** [review]
Thanks @pablo-de-andres - this LGTM as high-level documentation!

Note that I have not actually tested the primary / dependent registration myself, though I'd be happy to try if there are any low-level instructions available (i.e how to specify the primary app in the dependent's OpenAPI file, etc..)

However, I expect this will be covered by the [private repo issue](https://gitlab.cc-asp.fraunhofer.de/MarketPlace/platform/issue-reporting/-/issues/111)

**Comment by pablo-de-andres** [comment]
There aren't any particular instructions, currently you would register all like normal apps, use the OAuth info of the primary app for the platform, and simply ignore that of the dependent apps.

The proper info and procedure will indeed be tackled by said issue.

I will merge for now just so the concept is part of the docs.

**Pull request merged**